processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -263,4 +264,24 @@ public static Float convertObjectToFloat(Object valObj)
       throw new ParseException("Unknown type[%s]", valObj.getClass());
     }
   }
+
+  public static Long getIntegralFromFloatString(String floatStr)
+  {
+    BigDecimal convertedBD;
+    try {
+      convertedBD = new BigDecimal(floatStr);
+    } catch (NumberFormatException nfe) {
+      return null;
+    }
+
+    long asLong;
+    try {
+      asLong = convertedBD.longValueExact();
+    } catch (ArithmeticException ae) {
+      // indicates there was a non-integral part, or the BigDecimal was too big for a long
```

Comment:
It seems to make sense, but during this conversation, I realized that it's not usual to cast floating points to integrals for comparison. As you know, usually casts are done in opposite direction during comparison. So, how about changing the type of lower/upper bounds from longs to floating points? I think it will be better because we can avoid the problem of handling non-integral part and also don't have to consider select/in filters here.